app/gui2/src/components/ToggleIcon.vue

```diff
@@ -0,0 +1,20 @@
+<script setup lang="ts">
+import SvgIcon from '@/components/SvgIcon.vue'
+import { ref } from 'vue'
+
+const props = defineProps<{ icon: String; modelValue?: Boolean }>()
+const emit = defineEmits<{
+  (e: 'update:modelValue', toggledOn: Boolean): void
+}>()
+
+const toggledOn = ref(props.modelValue ?? false)
+
+function toggle() {
+  toggledOn.value = !toggledOn.value
+  emit('update:modelValue', toggledOn.value)
+}
+</script>
+
+<template>
+  <SvgIcon :variant="props.icon" :class="{ toggled_on: toggledOn }" @click="toggle" />
```

Comment:
fwiw this is what i did for the checkbox component. no clue whether it's conventional vue though.

https://github.com/enso-org/enso/blob/81bdf3fb463f00ed72da0d993b99e618b604a236/app/gui2/src/components/widgets/Checkbox.vue#L1-L10